src/scripts/ui/inline-roll-links.ts

```diff
@@ -184,7 +184,7 @@ export const InlineRollLinks = {
                             }
 
                             const targetActor = pf2Defense
-                                ? targetOwner
+                                ? targetOwner === "true"
```

Comment:
This will never be the case

Reply:
how sure are you about this?
Because I tested it while writing the code.
![image](https://github.com/foundryvtt/pf2e/assets/74384067/aa9a8f16-804f-43f9-b2b7-95bac82fe74a)
![image](https://github.com/foundryvtt/pf2e/assets/74384067/193c63ea-f3af-4ac1-b894-8f3eddc42c13)
![image](https://github.com/foundryvtt/pf2e/assets/74384067/089d7703-ab15-44c7-90af-9eed14b6174b)

Reply:
whoops

the bug is instead in `createHTMLElement`. Pop this in:

```diff
diff --git a/src/util/dom.ts b/src/util/dom.ts
index c2287df520..6ec7268361 100644
--- a/src/util/dom.ts
+++ b/src/util/dom.ts
@@ -35,8 +35,8 @@ function createHTMLElement<K extends keyof HTMLElementTagNameMap>(
     const element = document.createElement(nodeName);
     if (classes.length > 0) element.classList.add(...classes);
 
-    for (const [key, value] of Object.entries(dataset).filter(([, v]) => !R.isNil(v))) {
-        element.dataset[key] = String(value);
+    for (const [key, value] of Object.entries(dataset).filter(([, v]) => !R.isNil(v) && v !== false)) {
+        element.dataset[key] = value === true ? "" : String(value);
     }
 
     if (innerHTML) {
```

Reply:
better for sure - I get `""` evaluated as falsy, but I can check against !== undefined, probably

Reply:
That's what the `"string" in dataset` does

Reply:
I'm not sure if I need to tweak something and where 😓
"string" in dataset only rings a bell for `"targetOwner" in rawParams`. This evaluates as true or false
Then the dom.ts makes it that the dataset either doesn't have data-target-owner or to have data-target-owner=""
when it's read later in the code from destructuring the dataset I get either `undefined` or `""` and both evaluate falsy in the `targetOwner ? a : b` syntax

I feel like I have four options here
1. set the value for true as String(value) in createHTMLElement, I get `"true"` and can do a simple `targetOwner ? a : b`
2. somehow make the dataset understand that a value of `""` should be represented as `true`, this seems like a bad idea since I'd have to change the type for link.dataset or add a new helper function with a more relaxed type
3. instead of using a destructuring assignment from the dataset, write a line `const targetOwner = "targetOwner" in link.dataset`
4. check if targetOwner is undefined later in the code.

I'll push up 3, writing this out it's probably what you meant